cmd/soroban-cli/src/commands/keys/generate.rs

```diff
@@ -60,17 +100,33 @@ impl Cmd {
         } else {
             seed_phrase
         };
-        self.config_locator.write_identity(&self.name, &secret)?;
-        if !self.no_fund {
-            let addr = secret.public_key(self.hd_path)?;
-            let network = self.network.get(&self.config_locator)?;
-            network
-                .fund_address(&addr)
-                .await
-                .map_err(|e| {
-                    tracing::warn!("fund_address failed: {e}");
-                })
-                .unwrap_or_default();
+
+        // Check if identity exists
+        if let Ok(existing_secret) = self.config_locator.read_identity(&self.name) {
```

Comment:
```suggestion
        let Ok(existing_secret) = self.config_locator.read_identity(&self.name) else {
            // Identity doesn't exist, create new one
            self.write_and_fund_identity(&secret).await?;
            return Ok(());
        };
```
Then  the stuff currently indented in the `if` can be shifted over for better readability.